sdk-base/src/main/java/com/mapbox/maps/threading/AnimationThreadController.kt

```diff
@@ -0,0 +1,120 @@
+package com.mapbox.maps.threading
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.logW
+
+/**
+ * Experimental controller singleton allowing to use background thread for all Mapbox animations.
+ * Most likely should not be used directly, calling [useBackgroundThread] should do all the work.
+ *
+ * By default all Mapbox animations are running on Android Main thread.
+ *
+ * Following Mapbox plugins are updated to make use of background animation thread:
+ *  - plugin-locationcomponent
+ *  - plugin-animation
+ *  - plugin-viewport
+ *
+ *  Switching between [useBackgroundThread] and [useMainThread] could be done at any point in runtime.
+ *
+ *  Using this class is extremely unsafe and error prone, it should be used in very specific use-cases!
+ *  Even when it's essential to use it (for example Android Auto use-case on specific devices when locking the device)
+ *  it would be safer to use Android Main thread as much as possible and switch to background thread
+ *  only when it's needed.
+ */
+@MapboxExperimental
+object AnimationThreadController {
```

Comment:
@kmadsen I agree and spent some time to prototype it however it complicates the approach dramatically. I don't think we need to do it now (we can add this later if needed) because:
1. `AnimationThreadController` is located in `sdk-base` module while `MapView`, `MapboxMap` etc are located in `sdk` module and `sdk-base` does not depend on `sdk`. We can't simply move `AnimationThreadController` to `sdk` module as our plugins depend on `sdk-base` and not `sdk`. Of course it still could be done but it's a bit of a burden.
2. Background thread should be used only for Android Auto + Xiaomi + main map gone to background use-case. For this case it does not actually matter that we change thread for all map instances as we do really care only about one. Using background thread in some other use-cases apart from this is high risk and does not provide any performance gain.

As said above - we could always revisit this later if this will be needed badly, all new APIs are experimental and we may break them if needed.

Reply:
> Background thread should be used only for Android Auto + Xiaomi + main map gone to background use-case.

This is a point that is a reason to bound the animation thread to android auto though. Right now as a singleton, the car will change both the car and phone (and watch one day?)

> AnimationThreadController is located in sdk-base module while MapView, MapboxMap etc are located in sdk module and sdk-base does not depend on sdk

Am unsure why this is a restriction. `MapView` and `MapboxMap` can see `AnimationThreadController` so they should be able to control the lifecycle 🤔 

> As said above - we could always revisit this later if this will be needed badly, all new APIs are experimental and we may break them if needed.

👍 I'm also onboard with moving forward with `AnimationThreadController` as a singleton for now.

The issue that we will face immediately though, is we need a stable api. For that we can move `CarAnimationThreadController` into the SDK and be able to declare it stable some time soon. It can surface the functionality and allow the `AnimationThreadController` change over time.